packages/server/src/core/parser.ts

```diff
@@ -18,11 +18,17 @@ export type ParserCustomValidatorEsque<TInput> = (
 export type ParserYupEsque<TInput> = {
   validateSync: (input: unknown) => TInput;
 };
+
+export type ParserScaleEsque<TInput> = {
+  assert(value: unknown): asserts value is TInput;
+};
```

Comment:
It allows transforms right? If so, it should be in `ParserWithInputOutput` with the possible input and output separated

Reply:
Can you please clarify what you mean by transforms?

Reply:
It supports transforms with regard to the `encode`/`decode`, but not in the `assert` logic.